File: quarkus-jdbi/src/main/java/com/github/frederikpietzko/dto/VisitDto.java
package com.github.frederikpietzko.dto;

import com.github.frederikpietzko.model.Visit;
import jakarta.validation.constraints.NotEmpty;
import jakarta.validation.constraints.NotNull;

import java.time.OffsetDateTime;

public record VisitDto(
  @NotNull Long id,
  @NotEmpty String description,
  @NotNull PetDto pet,
  @NotNull OffsetDateTime dateTime
) {

  public static VisitDto fromEntity(Visit entity) {
    return new VisitDto(
      entity.getId(),
      entity.getDescription(),
      PetDto.fromEntity(entity.getPet()),
      entity.getDateTime()
    );
  }
}

File: ktor/src/main/kotlin/com/github/frederikpietzko/dto/VisitDto.kt
package com.github.frederikpietzko.dto

import kotlinx.serialization.Serializable
import java.time.OffsetDateTime

@Serializable
data class VisitDto(
  val id: Long,
  val description: String,
  val pet: PetDto,
  @Serializable(with = OffsetDateTimeSerializer::class) val dateTime: OffsetDateTime,
)

fun com.github.frederikpietzko.model.Visit.toDto() = VisitDto(
  id = id!!,
  description = description,
  pet = pet.toDto(),
  dateTime = dateTime,
)
